### ISSUE TITLE
Parallelize BIDS2Legacy: x.D.ROOT usage

### ISSUE THREAD

**Issue by jan-petr** [ExploreASL/ExploreASL#1547]
### Description

This request started as wanting to parallelize BIDS2Legacy because it conflicted and took too long for processing (a lot of overhead.

To do this properly, this changed into changing the data loading, as follows:

* Import has nothing to do with BIDS2Legacy, or data loading
* At the start of processing, we load data, either from rawdata (BIDS2Legacy) or from derivatives (pre-existing)
* Then we define parallelization
* Then we define all other metadata accordingly

### Tasks
- [x] `x.D.ROOT` -> replaceD by `x.dir.RawData` or `x.dir.xASLDerivatives`
- [X] xASL_init_BIDS2Legacy & xASL_module_BIDS2Legacy use legacy format (subject_visit)
- [X] x.opts.bSkipBIDS2Legacy fully removed
- [x] Bugfix potential participants.tsv creation in parallel
- [x] MANAGE participants.tsv
xASL_init_DefineStudyData -> xASL_init_LoadMetadata -> xASL_bids_LoadParticipantsTSV
- [x] MANAGE dataset_description.tsv — when loading data
       These BIDS files should be in rawdata, and copied/merged to xASLDerivatives if there is not one already
       If this file exists in the datasetroot, issue a warning that we ignore this one
xASL_init_BIDS2Legacy -> checks availability dataset_description.json in rawdata
xASL_bids_CompleteBIDS2Legacy -> copies dataset_description from rawdata to derivatives
- [x] MANAGE DATAPAR — when starting processing (not needed for loading data, this is a processing settings file, but 
       can do before loading as well)
       This ExploreASL-specific settings file should be in datasetroot and copied to xASLDerivatives if there is not one 
       already
       If this file exists in rawdata, issue a warning  that we ignore this one
       xASL_init_BIDS2Legacy section 3 -> 
       xASL_bids_CompleteBIDS2Legacy 30 -> create dataPar if not existed
       xASL_init_DataLoading 372 -> read dataPar (and deal with backward compatibility)
- [x] MANAGE subjectRegexp
- [x] Major revamp initialization structure
- [x] Update headers

#### Fixes done for testing Jan
- [x] Revamp the order
Changes:
* Move xASL_init_LoadDataPar & xASL_init_RemoveLockDirs from AFTER subject loading to BEFORE subject loading
* Move xASL_init_PrintUserFeedback from AFTER subject loading to BEFORE subject loading
* Exchange names between xASL_init_DataLoading (BEFORE subject loading) & xASL_init_Process (AFTER subject loading)

Old order:

xASL_init_Process

    xASL_init_SubjectList
    xASL_init_Parallelization
    xASL_module_BIDS2Legacy
    xASL_init_CreateParticipantsTSV
    xASL_init_Session_TimePoint_Lists

xASL_init_DataLoading
>xASL_init_LoadDataPar
> xASL_init_Loaddataset_description
> xASL_init_DefineDataParDependentSettings
> xASL_init_DefineStudyStats
> xASL_init_RemoveLockDirs
> xASL_init_PrintCheckSettings

xASL_init_PrintUserFeedback

New order:

xASL_init_DataLoading

>xASL_init_LoadDataPar
> xASL_init_RemoveLockDirs

xASL_init_PrintUserFeedback

    xASL_init_SubjectList
    xASL_init_Parallelization
    xASL_module_BIDS2Legacy
    xASL_init_CreateParticipantsTSV
    xASL_init_Session_TimePoint_Lists

xASL_init_Process
> xASL_init_Loaddataset_description
> xASL_init_DefineDataParDependentSettings
> xASL_init_DefineStudyStats
> xASL_init_PrintCheckSettings

       

### Testing fixes non-specific to this issue
- [x] xASL_im_CreateGroupAnalysisMask: bugfix loading paths instead of cells
- [x] xASL_wrp_CreatePopulationTemplates: reduce verbosity for non-existing output
- [x] xASL_qc_CompareTemplate: bugfix too slow pinv
- [x] xASL_io_CheckDeprecatedFieldsX: only check existing fields

#### PM
HENK: I removed 2 obsolete tasks here. After you read this, you can remove this line :)
- [x] Move ExploreASL/Functions/Import -> ExploreASL/Modules/Module_Import JAN: Can be done now.
       HENK: done
- [x] Separated ExploreASL/Functions/BIDS into 
        ExploreASL/Modules/Module_Import/NII2BIDS 
        ExploreASL/Modules/Module_Processing/Module_BIDS2Legacy
        Testing/BIDS
### MOVED TO #1275 
- [x] What should happen with /Functions/BIDS/xASL_bids_CheckDatasetDescription.m? This file is currently not used.
- [x] Also, in ExploreASL_Initialize -> xASL_init_AddDirsOfxASL we now initialize all folders used in import, processing, testing. I think it would be nicer to move the import folders to the import initialization, the processing folders to the processing initialization, and the testing folders to the testing, don't you agree? Then we are sure that we are never mixing up code. But we can't really do this yet, because there are a few things we use for both import and processing. But we could already do this for testing right? The users don't need to see the testing folder?

- [x] Adapt documentation: remove old and add new parameters JAN: Are there any new/old parameters? Which ones?
HENK:
x.opts.bSkipBIDS2Legacy is removed
x.dataset.TimePoint*: some of these fields might be removed but these are internal fields only I guess?

- [x] BIDS2Legacy should use the rawdata subjects (ignore the derivatives subjects). JAN: Agree, and it is possible to run this with purely rawdata input without derivatives and sourcestructure, so that should work.

- [x] If we define subjects (`x.opts.subjectFolder`) from `x.dir.RawData`, then the problem is that our legacy folder doesn't have the BIDS names yet. So we should fix that (i.e., adding the _1 suffix).
<img width="919" alt="image" src="https://github.com/ExploreASL/ExploreASL/assets/27774254/b0b91a22-34a0-4aac-b0b2-7ccb4c202e49">

### How to test
- [x] Flavors
- [x] TestDataSets
- [x] Update unit tests & test

### Release notes

Report with https://github.com/ExploreASL/ExploreASL/issues/1275. Allow parallelization of BIDS2Legacy.
